### ISSUE TITLE
Intellij rust plugin does not give warning when variable does not need to be mutable

### ISSUE THREAD

**Issue by Haider8** [intellij-rust/intellij-rust#5307]
## Environment

* **IntelliJ Rust plugin version:** 0.2.121.3059-201-stable
* **Rust toolchain version:** rustc 1.43.0 (4fb7144ed 2020-04-20)
* **IDE name and version:** CLion-2020.1
* **Operating system:** Linux (16.04.1-Ubuntu)

## Problem description
It does not give a warning if a variable does not need to be mutable like the rust compiler gives when you compile your program like this `warning: variable does not need to be mutable
`

## Steps to reproduce
Write this program :
```
fn main() {
    let mut s = String::from("hello world");
    println!("String is {}", s);
}
```
Rust plugin will not warn or give a hint to remove `mut` keyword. Whereas, the compiler will give you this warning :
```
warning: variable does not need to be mutable
 --> src/main.rs:2:9
  |
2 |     let mut s = String::from("hello world");
  |         ----^
  |         |
  |         help: remove this `mut`
  |
  = note: `#[warn(unused_mut)]` on by default
```

**Comment by Kobzol** [comment]
Duplicate of https://github.com/intellij-rust/intellij-rust/issues/1004.
